code/processes/dqe.q

```diff
@@ -24,54 +26,101 @@ fillprocname:{[rs;h]
+runsetup:{[idnum;fn;vars;rs]
   fncheck:` vs fn;
   if[not fncheck[2] in key value .Q.dd[`;fncheck 1];                                                            /- run check to make sure passed in function exists
     .lg.e[`runcheck;"Function ",(string fn)," doesn't exist"];
     :()];
 
+  h:.dqe.gethandles[rs]                                                                                         /- check if processes exist and are valid
+  }
+
+runcheck:{[idnum;fn;vars;rs]                                                                                    /- function used to send other function to test processes
   rs:(),rs;                                                                                                     /- set rs to a list
-  h:.dqe.gethandles[rs];                                                                                        /- check if processes exist and are valid
+  h:.dqe.runsetup[idnum;fn;vars;rs];
 
   r:.dqe.fillprocname[rs;h];
   .dqe.initstatusupd[idnum;fn;vars]'[r];
-  
-  .dqe.failchk[idnum;"error:can't connect to process";`];
+
+  .dqe.updresultstab[idnum;0Np;0b;"error:can't connect to process";`failed;0b;`];
   procsdown:(h`procname) where 0N = h`w;                                                                        /- checks if any procs didn't get handles
-  if[count procsdown;.dqe.failchk[idnum;"error:process is down or has lost its handle"]'[procsdown]];
+  if[count procsdown;.dqe.updresultstab[idnum;0Np;0b;"error:process is down or has lost its handle";`failed;0b]'[procsdown]];
 
   if[0=count h;.lg.e[`handle;"cannot open handle to any given processes"];:()];                                 /- check if any handles exist, if not exit function
-  ans:.dqe.getresult[value fn;(),vars;idnum]'[h[`procname];h[`w]]
+  ans:.dqe.getresult[value fn;(),vars;idnum;(),0b]'[h[`procname];h[`w]]
+  }
+
+runcomparison:{[idnum;fn;vars;rs;comptype;compproc]                                                             /- ran for comparison checks between multiple processes
+  rs:(),rs;                                                                                                     /- set rs to a list
+  h:.dqe.runsetup[idnum;fn;vars;rs];
+
+  if[compproc in h`procname;                                                                                    /- fail if comparison process is in list of processes to check against
+    .dqe.updresultstab[idnum;0Np;0b;"error:compare process can't be compared with itself";`failed;0b]'[h`procname];
+    :()];
+
+  comph:.dqe.gethandles[compproc];                                                                              /- obtain handle for comparison process
+  h:h,'comph;
+  proccount:count h`procname;
+
+  r:.dqe.fillprocname[rs;h];
+  .dqe.initstatusupd[idnum;fn;vars]'[r];
+
+  if[(` in r[;1])|(0N in h`w);
```

Comment:
If it's just nulls you are looking for then it may be clearer to use `any null`:
```suggestion
  if[any[null h`w]|any null r[;1]
```